### DISCUSSION TITLE
Cross-origin isolation

### DISCUSSION THREAD

**Discussion by ChronoDK** [fastapi/fastapi#3077] [Questions]
Hi,

I'm hosting a game that makes use of SharedArrayBuffer, which will soon require cross-origin isolation:
https://blog.chromium.org/2021/02/restriction-on-sharedarraybuffers.html

```
Cross-Origin-Embedder-Policy: require-corp
Cross-Origin-Opener-Policy: same-origin
```

How do I configure this in FastAPI?

The game is just an HTML file and some resources, shared like this:
`@app.get("/", response_class=HTMLResponse)`

**Comment by ChronoDK** [comment]
I could not get CORSMiddleware to work at all - might be broken somehow?

This works though, incase anyone needs it later:

```@app.get("/", response_class=HTMLResponse)
async def game_index(request: Request):
    @app.middleware("http")
    async def add_cors_header(request: Request, call_next):
        response = await call_next(request)
        response.headers["Cross-Origin-Embedder-Policy"] = "require-corp"
        response.headers["Cross-Origin-Opener-Policy"] = "same-origin"
        return response

    response = templates.TemplateResponse("index.html", {"request": request, "projectName": PROJECT_NAME})
    response.headers["Cross-Origin-Embedder-Policy"] = "require-corp"
    response.headers["Cross-Origin-Opener-Policy"] = "same-origin"
    return response

**Reply by Mause** [reply]
The CORS middleware doesn't support those headers - they're part of the CORP specification, not the CORS specification

**Reply by ChronoDK** [reply]
Thanks, I guess there are things I need to read up on :-D
